src/gc/unix/gcenv.unix.cpp

```diff
@@ -318,6 +330,28 @@ bool GCToOSInterface::VirtualRelease(void* address, size_t size)
     return (ret == 0);
 }
 
+// Commit virtual memory range.
+// Parameters:
+//  size      - size of the virtual memory range
+// Return:
+//  Starting virtual address of the committed range
+void* GCToOSInterface::VirtualReserveAndCommitLargePages(size_t size)
+{
+#ifdef __linux__
+    uint32_t largePagesFlag = MAP_HUGETLB;
```

Comment:
>largePagesFlag [](start = 13, length = 14)

do we not want to specifically say MAP_HUGETLB | MAP_HUGE_2MB or MAP_HUGETLB | MAP_HUGE_1GB here?

Reply:
I don't think we want to hardcode the huge page size as we don't know what huge page sizes are supported on the target system. We may want to add (now or later) a new config to allow the user to select the huge page size wanted. The number of supported huge page sizes is not fixed, there can be more than 2MB and 1GB. The supported sizes can be found by listing directories and parsing their names in the `/sys/kernel/mm/hugepages/` (that's the official way of finding this out based on the mmap doc). There is one directory per supported huge page size there. On my system, I have `hugepages-2048kB` there.